leap/rk/__init__.py

```diff
@@ -203,166 +258,157 @@ def generate_butcher(self, stage_coeff_set_names, stage_coeff_sets, rhs_funcs,
 
         knowns = set()
 
-        # {{{ stage loop
-
-        last_state_est_var = cb.fresh_var("last_state_est")
-        last_state_est_var_valid = False
-
-        with CodeBuilder(name="primary") as cb:
-            equations = []
-            unknowns = set()
-
-            def make_known(v):
-                unknowns.discard(v)
-                knowns.add(v)
-
-            for istage in range(nstages):
-                for name in stage_coeff_set_names:
-                    c = self.c[istage]
-                    my_rhs = stage_rhs_vars[name][istage]
-
-                    if (
-                            name in self.recycle_last_stage_coeff_set_names
-                            and istage == 0
-                            and _is_first_stage_same_as_last_stage(
-                                self.c, stage_coeff_sets[name])):
-                        cb(my_rhs, last_rhss[name])
-                        make_known(my_rhs)
+        def make_known(v):
+            unknowns.discard(v)
```

Comment:
Why is this here? Is there ever a scenario where a variable would be added to `unknowns` only to have to be removed later?

Reply:
This is something that I pulled in from the existing Butcher generator. Truthfully I'm not sure that such a scenario exists, though based on where this function is called, it looks like it might be possible in the case of stage recycling. Let me run some tests to determine whether this is even needed, because it's a fair question.

Reply:
Addressed in 61277d8. I've removed this function and replaced calls to it with adding rhs's to knowns, but I agree with you where we should never need to remove things from unknowns, and getting rid of that has no ill effect that I can see.